src/main/java/ladder/controller/LadderController.java

```diff
@@ -1,41 +1,130 @@
 package ladder.controller;
 
-import java.util.function.Function;
-import java.util.function.Supplier;
+import java.util.List;
+import java.util.Optional;
 import ladder.domain.Height;
 import ladder.domain.Ladder;
 import ladder.domain.Players;
-import ladder.domain.generator.LadderGenerator;
-import ladder.domain.generator.LineGenerator;
+import ladder.domain.Prizes;
+import ladder.domain.Result;
+import ladder.domain.Retry;
+import ladder.domain.generator.RandomDirectionGenerator;
 import ladder.view.InputView;
 import ladder.view.OutputView;
 
 public class LadderController {
 
+    private static final String ALL_COMMAND = "all";
+
     private final InputView inputView;
     private final OutputView outputView;
-    private final LadderGenerator ladderGenerator;
 
-    public LadderController(final InputView inputView, final OutputView outputView,
-                            final LadderGenerator ladderGenerator) {
+    public LadderController(final InputView inputView, final OutputView outputView) {
         this.inputView = inputView;
         this.outputView = outputView;
-        this.ladderGenerator = ladderGenerator;
     }
 
     public void run() {
-        final Players players = generate(inputView::readPlayerNames, Players::new);
-        final Height height = generate(inputView::readHeight, Height::new);
-        final Ladder ladder = ladderGenerator.generate(new LineGenerator(), players, height);
-        outputView.printLadderResult(players, ladder);
+        final Players players = readPlayers(new Retry(5));
+        final Prizes prizes = readPrizes(new Retry(5), players);
+        final Height height = readHeight(new Retry(5));
+        final Ladder ladder = Ladder.of(new RandomDirectionGenerator(), players, height);
+        outputView.printLadderGame(players, ladder, prizes);
+        final Result result = Result.of(players, ladder, prizes);
+        final String target = readTarget(new Retry(5), result);
+        printResult(result, target);
+    }
+
+    private Players readPlayers(final Retry retry) {
+        checkCount(retry);
+        final Optional<Players> players = generatePlayers();
+        if (players.isEmpty()) {
+            retry.decrease();
+            return readPlayers(retry);
+        }
+        return players.get();
+    }
```

Comment:
재귀함수는 지양해 주세요 :) 

```suggestion
private Players readPlayers(final Retry retry) {
        while (retry.isPossible()) {
            var players = generatePlayers();
            if (players.isPresent()) {
                return players.get();
            }
            retry.decrease();
        }
        throw new IllegalStateException("재입력 횟수를 초과하였습니다.");
    }
```

2 depth정도는 괜찮습니다 :)

Reply:
재귀 함수 사용을 지양하려고 했으나 이번 미션 프로그래밍 요구사항에 `depth는 1까지 허용한다`라고 되어 있어 사용했습니다 😂

재입력 일정 횟수가 존재하면서 depth 1을 지키기 위해서는 재귀를 사용하지 않고는 해결할 수 없었습니다 .. 요구사항을 조금 벗어나는 반란(포비가 강조하는)을 해도 괜찮을까요?

Reply:
넵..! depth 1을 만들기위해 로직이 더 복잡해지고 어려워진다면, 과감히..조금 벗어나도 괜찮을 것 같아요! (물론 저의 개인적인 생각입니다... 저는 depth2 까지는 작성했었던것 같아요 :) )